### ISSUE TITLE
Pre-trained concept embedding seems barely improve the model performance?

### ISSUE THREAD

**Issue by 1230pitchanqw** [mp2893/doctorai#2]
Hi Edward,

As I am reading some nlp papers about using pre-trained embedding can largely improve the downstream nlp tasks, like NMT. I am wondering is it still true that the pre-trained concept embedding can improve the healthcare tasks, like disease prediction in your paper.

It seems that in your experiment, the pre-trained concept embedding barely improve the model performance (only 1% increase of the recall rate), while the pre-trained RNN on scutter dataset clearly largely improve the model performance on MIMIC dataset.

So, why the pre-trained concept embedding seems barely improve the model performance

Is it because:
1. the quality of the pre-trained embedding is not good enough as data source is relatively small, skip-gram will suffer from the zipfian's law. 
2. there is an aggregate operation on the visit-level, which sums all the medical code embedding and hence downplay the influence of the medical code embedding.
3. the characteristic of the task itself. There are two many new diseases that are unpredictable, with 70% recall rate is already reaching the max performance.

thank you so much!
Xianlong

**Comment by mp2893** [comment]
Hi Xianlong,

You raise an important question, and your answers are all relevant to some extent.
Based on my experience (mostly from experiments I ran for several papers), the general rule of thumb for using the pre-trained embedding is as follows:
- Pre-training your medical concept embeddings and fine-tuning them during the downstream prediction task should demonstrate at least the same level of performance as when you don't pre-train your embeddings. 
- The benefit of pre-training the embeddings varies depending on your task, dataset size, and model architecture. 
  -- The more data you have, the smaller the benefit of using pre-trained embeddings will be. 
  -- In my experience, you will benefit more from using pre-trained embeddings when your task requires correlation information between medical concepts, because you typically pre-train your embeddings using cooccurrence-based algorithms For example, DoctorAI tries to predict all codes occurring in the next visit, hence can benefit from code correlation information. On the other hand, when you have an external label (e.g. mortality, length of stay), then the benefit of using pre-trained embeddings diminishes. But this is coming from very limited experience so this may not generalize.
  -- As you pointed out, summing the embeddings at the visit-level might cancel out the embedding values. This is one of the reasons I used ReLU at the code-level embedding in Med2Vec. But it seems neural network can learn whatever you tell it to learn, regardless of the summation at the visit-level, because after Med2Vec, I didn't particularly stick to ReLU at the code-level. Of course, a rigorous experiment to answer this question can be an interesting topic. (But I doubt there will be significant difference, because I've seen NLP studies where they sum the word embeddings to derive sentence representation and they seemed to work fine).
- Many unsupervised representation learning algorithms use co-occurrence information of concepts (medical concept or words), and I suspect using co-occurrence information can only go so far. Maybe you can come up with a novel representation learning approach and help many researchers out there :)

Hope this helps
Ed

**Comment by 1230pitchanqw** [comment]
Hi Edward,

Really appreciate your response! It is very helpful!

I ran some disease prediction experiments with mimic-3 dataset using the same model (RNN) and hyper-parameters (2000 RNN hidden units, 200 embedding size and ect.) as you did in the paper.

- Without pre-trained embedding, the recall@30 is around 52.0%. (similar to the results in your paper)
- With skip-gram pre-trained (on mimic-3), the recall raise to 56.0%.
- It is interesting that when I use your med2vec pre-trained embedding (1/3 code initialized randomly as not appear in the med2vec vocabulary), the recall rate is only 53.5%. 

Since med2vec is using a much larger dataset, I would expect it has a greater influence on the disease prediction task. 

I think it might because of heterogeneity of medical codes? (i.e. codes in children dataset might have different "meaning" to codes in ICU, and their co-occurrence information will then be different). What you you think?

Thanks!

**Comment by mp2893** [comment]
Hi Xianlong, 

First of all I have to ask:
When you pre-trained your embeddings on mimic-3, did you pre-train the embeddings on the entire mimic-3, or just the training set you used to train the disease prediction model?
If it's the former, then you are spilling information from the test set indirectly into the model, so comparing that to med2vec embeddings wouldn't be fair.

That being covered, I agree with your point; Med2vec embeddings on my webpage were trained on records from children's hospital only, so the diagnosis/medication/procedure distribution would be very different from MIMIC-3, which consists of ICU records. But still, it should be better than nothing (as demonstrated by your experiments), because MIMIC-3 is such a small dataset (only 7~8k patients who made at least two visits).

Best,
Ed

**Comment by 1230pitchanqw** [comment]
Hi Ed,
Sorry, I somehow missed your comment.
You are right! I did pre-train on the entire dataset instead of the training dataset. 
Pre-train only on training gives me a 53% recall rate. 
Thanks for your valuable comment!
